scripts/docker-build.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+set -eEuo pipefail
+
+#############################################
+# USAGE                                     #
+#############################################
+
+function title() {
+  echo "Orbit reproducible builds"
+}
+
+function usage() {
+  cat <<EOF
+
+Usage:
+  $0
+
+Options:
+  --control-panel builds the control panel canister
+  --station builds the station canister
+  --upgrader builds the upgrader canister
+
+  -h, --help prints this help message
+EOF
+}
+
+function help() {
+  cat <<EOF
+
+Helper script to facilitate the building of the canisters in the Orbit project in a reproducible way.
+
+NOTE: This requires a working Docker installation.
+EOF
+}
+
+function exec_function() {
+  local function_name=$1
+  shift
+  echo "------------------------------------------------------"
+  echo -e "\e[1m$ START:\e[0m $function_name"
+  $function_name "$@"
+  echo -e "\e[1m$ COMPLETED:\e[0m $function_name"
+}
+
+#############################################
+# FEATURES                                  #
+#############################################
+
+function build_canister() {
+  local canister_name=$1
+  local target=$2
+
+  # Build the canister
+  docker build -t orbit-$canister_name --target $target .
+
+  # Create a container to extract the generated artifacts
+  docker create --name orbit-$canister_name-container orbit-$canister_name
```

Comment:
Is there any difference between these next lines and running it like this?

```
docker run --rm -v ./artifacts:/code/artifacts/$canister_name orbit-$canister_name-container
```

Reply:
With `-v` it is less clear that the Docker container doesn't just take your existing artifacts instead of building them from scratch.

Reply:
`docker cp` avoids potential file permission issues, that's the main reason i went for it.